include/sys/socket.h

```diff
@@ -314,10 +319,22 @@
 
 struct sockaddr_storage
 {
-  sa_family_t ss_family;     /* Address family */
-  char        ss_data[SS_MAXSIZE - sizeof(sa_family_t)];
-}
-aligned_data(SS_ALIGNSIZE);  /* Force desired alignment */
+  sa_family_t ss_family;       /* Address family */
+
+  /* Following fields are implementation-defined */
+
+  struct
+  {
+    char ss_pad1[SS_PAD1SIZE]; /* 6-byte pad; this is to make implementation-defined
+                                * pad up to alignment field that follows explicit in
+                                * the data structure */
+    int64_t ss_align;          /* Field to force desired structure storage alignment */
```

Comment:
The use of an **int64_t** type for the ss_align member in the sockaddr_storage structure can result in alignment problems that may prevent functions such as **getifaddrs(&ifap)** from correctly retrieving IP addresses. 
Here, should **int32_t ss_align**; be used or should the **structure sockaddr_storage** for the generic socket address be modified with **__attribute__((packed))**?
![image](https://github.com/user-attachments/assets/edd95d84-02e7-45a9-bce3-39cb2c7fb538)

Reply:
@pkarashchenko could you look at this problem?